pipeline/backend/kubernetes/pod.go

```diff
@@ -97,6 +102,10 @@ func Pod(namespace string, step *types.Step, labels, annotations map[string]stri
 
 	NodeSelector := map[string]string{"kubernetes.io/arch": strings.Split(platform, "/")[1]}
 
+	for key, val := range step.BackendOptions.Kubernetes.NodeSelector {
+		NodeSelector[key] = val
+	}
```

Comment:
While this appends to the previous `NodeSelector` object, it should be highlighted in the docs that `"kubernetes.io/arch"` is inferred from the top-level `platform` setting and all other options must be into `backend_options.NodeSelector`.

Reply:
I've updated the documentation